cli/lib/util.js

```diff
@@ -286,14 +288,29 @@ const util = {
     .value()
   },
 
-  getOriginalNodeOptions (options) {
+  getOriginalNodeOptions ({ processVersions }) {
+    const opts = {}
+
     if (process.env.NODE_OPTIONS) {
-      return {
-        ORIGINAL_NODE_OPTIONS: process.env.NODE_OPTIONS,
-      }
+      opts.ORIGINAL_NODE_OPTIONS = process.env.NODE_OPTIONS
+    }
+
+    // https://github.com/cypress-io/cypress/issues/18914
+    // Node 17+ ships with OpenSSL 3 by default, so we may need the option
+    // --openssl-legacy-provider so that webpack@4 can use the legacy MD4 hash
+    // function. This option doesn't exist on Node <17 or when it is built
+    // against OpenSSL 1, so we have to detect Node's major version and check
+    // which version of OpenSSL it was built against before spawning the plugins
+    // process.
+
+    // To be removed when the Cypress binary pulls in the @cypress/webpack-batteries-included-preprocessor
+    // version that has been updated to webpack >= 5.61, which no longer relies on
+    // Node's builtin crypto.hash function.
+    if (processVersions && semver.satisfies(processVersions.node, '>=17.0.0') && processVersions.openssl.startsWith('3.')) {
+      opts.ORIGINAL_NODE_OPTIONS = `${opts.ORIGINAL_NODE_OPTIONS || ''} --openssl-legacy-provider`
     }
 
-    return {}
+    return opts
```

Comment:
what does this return value get used for? Previously it always returned an empty object, but now we actually return stuff.

Reply:
previously in the `if process.env.NODE_OPTIONS ` check it'd return fast with
```
{ ORIGINAL_NODE_OPTIONS: ... }
```
This removed the fast-return to got through both checks before returning. If neither check are hit, it'll continue to return  an empty object.